iree/compiler/Dialect/Shape/Conversion/test/shape_to_shapex.mlir

```diff
@@ -45,6 +45,40 @@ func @f(%arg0: tensor<?xf32>, %index: i32) {
   return
 }
 
+// -----
+// shape.get_extent
+// CHECK-LABEL: func @f
+func @f(%arg0: tensor<?xf32>) {
+  %c0 = constant 0 : index
+  // CHECK: [[SHAPE:%.+]] = shapex.get_ranked_shape %arg0 : tensor<?xf32> -> !shapex.ranked_shape<[?]>
+  %0 = "shape.shape_of"(%arg0) : (tensor<?xf32>) -> !shape.shape
+
+  // CHECK: [[DIM:%.+]] = shapex.ranked_dim [[SHAPE]][0] : !shapex.ranked_shape<[?]> -> index
+  %result = shape.get_extent %0, %c0 : !shape.shape, index -> !shape.size
+  return
+}
+
+// -----
+// shape.from_extents
+// CHECK-LABEL: func @f
+func @f(%arg0: tensor<?x?xf32>) {
+  %c0 = constant 0 : index
+  %c1 = constant 1 : index
+
+  // CHECK: [[SHAPE:%.]] = shapex.get_ranked_shape %arg0 : tensor<?x?xf32>
```

Comment:
Can we just test
```
func @f(%arg0: index, %arg1: index) {
   %result = "shape.from_extents"(%arg0, %arg1) : (index, index) -> !shape.shape
  return
}
```

?

Reply:
Originally I wanted to check some of the intermediate types, moved to only testing with index.